modules/margin-protocol/src/lib.rs

```diff
@@ -508,11 +507,18 @@ impl<T: Trait> Module<T> {
 
 		Ok(())
 	}
+
+	/// Update `who` balance by `amount`.
+	///
+	/// Note this function guarantees op, don't use in possible no-op scenario.
+	fn _update_balance(who: &T::AccountId, amount: Fixed128) {
+		let new_balance = Self::balances(who).saturating_add(amount);
```

Comment:
use `mutate`

Reply:
`mutate` doesn't work for `Fixed128`, as they can't be derefered.

Reply:
@shaopengw @xlc maybe we can impl `Deref` for Fixed128

Reply:
@ermalkaleci  According to [`Deref` doc](https://doc.rust-lang.org/std/ops/trait.Deref.html)
> ...**`Deref` should only be implemented for smart pointers** to avoid confusion.